core/ui/src/main/kotlin/com/merxury/blocker/core/ui/applist/AppList.kt

```diff
@@ -57,13 +68,30 @@ fun AppList(
     onEnableClick: (String) -> Unit = {},
     onDisableClick: (String) -> Unit = {},
     onServiceStateUpdate: (String, Int) -> Unit = { _, _ -> },
+    onRefresh: () -> Unit = {},
```

Comment:
I don't think the refresh feature belongs to the List. You should move the refresh logic to the screen level.